catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java

```diff
@@ -1015,10 +1015,14 @@ public boolean dropTable(NameIdentifier tableIdent) {
    */
   @Override
   public boolean purgeTable(NameIdentifier tableIdent) throws UnsupportedOperationException {
-    if (isExternalTable(tableIdent)) {
-      throw new UnsupportedOperationException("Can't purge a external hive table");
-    } else {
-      return dropHiveTable(tableIdent, true, true);
+    try {
+      if (isExternalTable(tableIdent)) {
+        throw new UnsupportedOperationException("Can't purge a external hive table");
+      } else {
+        return dropHiveTable(tableIdent, true, true);
+      }
+    } catch (NoSuchTableException e) {
+      return false;
```

Comment:
@mchades can you please check this, is it the right fix, also I think we don't need to mix the changes here, if it is an issue we can use a separate PR to address it.

Reply:
yeah, #3685 will track this, those changes should be reverted in this PR